Assets/Scripts/Characters/Humans/Bullet.cs

```diff
@@ -214,75 +215,46 @@ private void killObject()
             UnityEngine.Object.Destroy(this.rope);
             UnityEngine.Object.Destroy(base.gameObject);
         }
-
-        public void launch(Vector3 v, Vector3 v2, string launcher_ref, bool isLeft, GameObject hero, bool leviMode = false)
+        
+        public void Launch(Source source, GameObject hookRef, Vector3 v, Vector3 v2, Hero hero, bool leviMode = false)
```

Comment:
I'm using a `Source` enum to carry the data about which transform the hook originated from, for example `hookRefL1`. This has the benefit of making it easier to create one authoritative `LaunchRope` method in `Hero`, and it allows us to send that enum rather than a `string` over the network in the `myMasterIs` RPC.